### PR TITLE
Adding unidentified issues modal

### PR DESCRIPTION
**Pull request by jiahuang**
Connects #6990

Also fixed a bug with remove issues.

add_unidentified_issue

Note that this PR has no backend changes. Backend changes will be discussed and opened up as another PR.

### CODE REVIEW

File: client/app/intake/actions/ama.js [hunk 1]
```diff
@@ -143,11 +143,23 @@ export const setIssueSelected = (profileDate, issueId, isSelected) => ({
...
   type: ACTIONS.REMOVE_ISSUE,
-  payload: { issue }
+  payload: { index }
 });
 
+export const addUnidentifiedIssue = (description, notes) => (dispatch) => {
+  dispatch({
+    type: ACTIONS.ADD_ISSUE,
+    payload: {
+      category: 'Unknown issue category',
```
Comment (jiahuang) [inline client/app/intake/actions/ama.js]: Should this be kept as "unknown issue category" for now to be consistent?

File: client/app/intake/reducers/common.js [hunk 2]
```diff
@@ -41,12 +50,14 @@ export const commonReducers = (state, action) => {
   };
 
   actionsMap[ACTIONS.REMOVE_ISSUE] = () => {
+    // issues are removed by position, because not all issues have referenceIds
```
Comment (leikkisa) [inline client/app/intake/reducers/common.js]: Don't they get a referenceId when you add it? formatAddedIssues makes it look like they all have them. I do think calling it referenceId is a little confusing though.

File: client/app/intake/util/index.js [hunk 3]
```diff
@@ -150,23 +150,26 @@ export const formatAddedIssues = (intakeData) => {
...
       return {
         referenceId: issue.id,
-        text: `${issue.category} - ${issue.description} Decision date ${formatDate(issue.decisionDate)}`
+        text: `${foundIssue.decision_text} Decision date ${formatDateStr(issue.profileDate)}.`,
+        notes: issue.notes
       };
     }
 
-    return {};
+    return {
```
Comment (leikkisa) [inline client/app/intake/util/index.js]: Maybe add a comment to indicate this for unrated issues.

File: client/app/intake/pages/addIssues.jsx [hunk 4]
```diff
@@ -32,15 +33,15 @@ class AddIssues extends React.PureComponent {
       return <div className="issues">
         <div>
           { issues.map((issue, index) => {
-            return <div className="issue" key={issue.referenceId}>
-              <div className="issue-desc">
-                <span className="issue-num">{index + 1}.</span>
+            return <div className="issue" key={`issue-${index}`}>
+              <div className={`issue-desc ${issue.isUnidentified}` ? 'unidentified-issue' : ''}>
+                <span className="issue-num">{index + 1}.&nbsp;</span>
```
Comment (leikkisa) [inline client/app/intake/pages/addIssues.jsx]: Nice fixes

File: client/app/intake/reducers/common.js (hunk 2 above)
Comment (jiahuang) [reply client/app/intake/reducers/common.js]: Reference ids are from rating issue reference ids, which don't exist for unrated/unidentified issues.

**Pull request merged**